superset-frontend/src/chart/chartAction.js

```diff
@@ -360,28 +360,33 @@ export function exploreJSON(
         // problem: response holds a list of results, when before we were just getting one result.
         // How to make the entire app compatible with multiple results?
         // For now just use the first result.
-        const result = response.result[0];
+        const queryResponse = response.result[0]; // deprecated
+        const queriesResponse = response.result;
 
-        dispatch(
-          logEvent(LOG_ACTIONS_LOAD_CHART, {
-            slice_id: key,
-            applied_filters: result.applied_filters,
-            is_cached: result.is_cached,
-            force_refresh: force,
-            row_count: result.rowcount,
-            datasource: formData.datasource,
-            start_offset: logStart,
-            ts: new Date().getTime(),
-            duration: Logger.getTimestamp() - logStart,
-            has_extra_filters:
-              formData.extra_filters && formData.extra_filters.length > 0,
-            viz_type: formData.viz_type,
-            data_age: result.is_cached
-              ? moment(new Date()).diff(moment.utc(result.cached_dttm))
-              : null,
-          }),
+        queriesResponse.forEach(resultItem =>
+          dispatch(
+            logEvent(LOG_ACTIONS_LOAD_CHART, {
+              slice_id: key,
+              applied_filters: resultItem.applied_filters,
+              is_cached: resultItem.is_cached,
+              force_refresh: force,
+              row_count: resultItem.rowcount,
+              datasource: formData.datasource,
+              start_offset: logStart,
+              ts: new Date().getTime(),
+              duration: Logger.getTimestamp() - logStart,
+              has_extra_filters:
+                formData.extra_filters && formData.extra_filters.length > 0,
+              viz_type: formData.viz_type,
+              data_age: resultItem.is_cached
+                ? moment(new Date()).diff(moment.utc(resultItem.cached_dttm))
+                : null,
+            }),
+          ),
+        );
+        return dispatch(
+          chartUpdateSucceeded(queryResponse, queriesResponse, key),
```

Comment:
What's the thinking behind adding another value here rather than replacing the current `queryResponse`? This is essentially doubling the data that's being passed around to components.

Reply:
hmm... good point, actually I wanted avoid refactor of a lot places in code, but now I see I hope I can handle it, I also will add some people to review, where I see my refactor affect there parts of code to be sure. I pushed now refactored code as you proposed